sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -714,6 +720,132 @@ private String toPrettyJson(Object object) {
         }
     }
 
+    /**
+     * Converts to an Object (only POJOs and JsonNode are supported).
+     *
+     * @param <T> the type of the object.
+     * @param c the class of the object, either a POJO class or JsonNode. If c is a POJO class, it must be a member
+     * (and not an anonymous or local) and a static one.
+     * @return the POJO.
+     * @throws IllegalArgumentException thrown if an error occurs
+     * @throws IllegalStateException thrown when objectmapper is unable to read tree
+     */
+    @SuppressWarnings("unchecked")
+    // Implicit or explicit cast to T is done after checking values are assignable from Class<T>.
+    public static <T> T toObjectFromObjectNode(JsonNode node, boolean isValueQuery, Class<T> c) {
+        // TODO: We have to remove this if we do not want to support InternalObjectNode anymore, and change all the
+        //  tests accordingly
+        if (InternalObjectNode.class.isAssignableFrom(c)) {
+            return (T) new InternalObjectNode((ObjectNode)node);
+        }
+
+        if (node instanceof ObjectNode &&
+            (JsonSerializable.class.isAssignableFrom(c) || containsJsonSerializable(c))) {
+
+            return c.cast(instantiateFromObjectNodeAndType((ObjectNode)node, c));
+        }
+
+        if (c == Short.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).shortValue();
+        }
+
+        if (Short.class.isAssignableFrom(c)) {
+            return c.cast((node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).shortValue());
+        }
+
+        if (c == Integer.TYPE) {
+            return (T)(Object)(node.isValueNode() ? node : node.get(Constants.Properties.VALUE)).intValue();
+        }
+
+        if (c == Integer.TYPE || Integer.class.isAssignableFrom(c)) {
```

Comment:
remove c == Integer.TYPE? (as there is one same check above)

Reply:
and what is the difference between these two check? c == Integer.TYPE, Integer.class.isAssignableFrom(c) => I guess my question is when it will happen it satisfies the first check, but not for the second one?

Reply:
Nope - c == Integer.TYPE is true for the primitive type (int) - the latter is for the boxed version java.lang.Integer

Reply:
got it, thanks